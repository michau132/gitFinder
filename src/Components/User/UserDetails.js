import React from 'react';
import PropTypes from 'prop-types';
import {
  Card, CardContent, Typography, CardActions, Button,
} from '@material-ui/core';
import styled from 'styled-components';

const StyledCard = styled(Card)`
  border: 2px solid black;
  width: 100%;
`;

const UserDetails = ({
  informations: {
    name, login, email, location,
  },
}) => (
  <StyledCard>
    <CardContent>
      <Typography variant="h6" component="h6" gutterBottom>{name}</Typography>
      <Typography variant="subtitle1">{login}</Typography>
      <Typography variant="subtitle1">{email}</Typography>
      <Typography variant="subtitle1">{location}</Typography>
    </CardContent>
    <CardActions>
      <Button
        href={`https://github.com/${login}`}
        variant="outlined"
        size="small"
      >
        Show profile
      </Button>
    </CardActions>
  </StyledCard>
);


UserDetails.propTypes = {
  informations: PropTypes.shape({
    login: PropTypes.string.isRequired,
    email: PropTypes.string,
    avatar_url: PropTypes.string,
    name: PropTypes.string,
  }).isRequired,
};

export default UserDetails;
