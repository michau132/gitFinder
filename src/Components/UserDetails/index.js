import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';

const UserDetails = ({ userInfo }) => (
  <section className={styles.userDetails}>
    <h2>Public data</h2>
    <h4>{userInfo.name}</h4>
    <h5>{userInfo.login}</h5>
    <p>
        Email:
      {userInfo.email}
    </p>
    <p>Location: {userInfo.location}</p>
    
  </section>
);


UserDetails.propTypes = {
  userInfo: PropTypes.shape({
    login: PropTypes.string.isRequired,
    email: PropTypes.string,
    avatar_url: PropTypes.string,
    name: PropTypes.string,
  }).isRequired,
};

export default UserDetails;
