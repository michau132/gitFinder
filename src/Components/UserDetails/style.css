.userDetails {
  border: 2px solid black;
  padding: 0 20px;
}