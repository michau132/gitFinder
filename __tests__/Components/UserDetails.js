import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserDetails from '../../src/Components/UserDetails';

describe('testing UserDetails component', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      userInfo: {
        name: 'Adam Oko',
        login: 'aOko123',
        email: 'aoko123@gmail.com',
        loctation: 'Raszyn',
      },
    };
    wrapper = shallow(
      <UserDetails {...props} />,
    );
  });
  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });
  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
