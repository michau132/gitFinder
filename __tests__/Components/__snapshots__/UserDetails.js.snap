// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserDetails component matching snapshot 1`] = `
<section>
  <h2>
    Public data
  </h2>
  <h4>
    Adam Oko
  </h4>
  <h5>
    aOko123
  </h5>
  <p>
    Email:
    aoko123@gmail.com
  </p>
  <p>
    Location:
  </p>
</section>
`;
